src/aws-cpp-sdk-core/include/aws/core/client/RequestCompression.h

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#pragma once
+
+#include <aws/core/Core_EXPORTS.h>
+#include <aws/core/client/ClientConfiguration.h>
+#include <aws/core/utils/Outcome.h>
+#include <aws/core/utils/memory/stl/AWSStreamFwd.h>
+#include <aws/core/utils/memory/stl/AWSString.h>
+#include <aws/core/utils/memory/stl/AWSVector.h>
+
+namespace Aws {
+namespace Client {
+
+enum class CompressionAlgorithm { NONE, GZIP };
+
+Aws::String GetCompressionAlgorithmId(const CompressionAlgorithm &algorithm);
+
+class AWS_CORE_API RequestCompression {
+public:
+  CompressionAlgorithm
+  selectAlgorithm(const Aws::Vector<CompressionAlgorithm> &proposedAlgorithms,
+                  const Aws::Client::RequestCompressionConfig &config,
+                  const size_t payloadLength);
+  Aws::Utils::Outcome<std::shared_ptr<Aws::IOStream>, bool>
```

Comment:
might worth aliasing to some CompressionOutcome type alias.

Reply:
done